nixos/lib/test-driver/test-driver.py

```diff
@@ -319,7 +319,11 @@ def wait_for_unit(self, unit, user=None):
     def get_unit_info(self, unit, user=None):
         status, lines = self.systemctl('--no-pager show "{}"'.format(unit), user)
         if status != 0:
-            return None
+            raise Exception(
+                'retrieving the info for unit "{}" {} failed with exit code {}'.format(
```

Comment:
maybe write "retrieving systemctl info for unit ..." would be more expressive for users who see this for the first time without consulting the source code of the driver.